sql/core/src/main/scala/org/apache/spark/sql/execution/command/SetCommand.scala

```diff
@@ -172,14 +172,19 @@ object SetCommand {
 case class ResetCommand(config: Option[String]) extends RunnableCommand with IgnoreCachedData {
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
-    val defaults = sparkSession.sharedState.conf
+    val globalInitialConfigs = sparkSession.sharedState.conf
+    val sessionDefaults = sparkSession.initialSessionOptions.filter {
+      case (k, _) => !SQLConf.staticConfKeys.contains(k)
```

Comment:
so the point here is not all `initialSessionOptions` are effective?  How can we tell which one is effective?

Reply:
Now there only one  `sharedState` created by the original session, the static SQL configs here clearly follow the copy in `sharedState`, and other dynamic SQL configs will respect the session's own `initialSessionOptions`.

In this case, the static ones in `initialSessionOptions` are ineffective. BUT it is hard to distinguish spark-core, runtime-SQL, and user ones,  the latter 2 will be reset and effective and the spark-core ones are not effective but will be reset.

Reply:
can we add a comment here saying that only non-static confs are effective?